### ISSUE TITLE
./macos-guest-virtualbox.sh: line 649: dmg2img: command not found

### ISSUE THREAD

**Issue by phobrla** [myspaghetti/macos-virtualbox#276]
Hi,

I am getting the error `./macos-guest-virtualbox.sh: line 649: dmg2img: command not found`

even though I have installed dmg2img at `"C:\cygwin64\dmg2img.exe"`

```
$ ./macos-guest-virtualbox.sh

Push-button installer of macOS on VirtualBox

This script installs only open-source software and unmodified Apple binaries,
and requires about 50GB of available storage, of which 25GB are for temporary
installation files that may be deleted when the script is finished.

The script interacts with the virtual machine twice, please do not interact
with the virtual machine manually before the script is finished.

Documentation about optional configuration, iCloud and iMessage connectivity,
resuming the script by stages, and other topics can be viewed with the
following command:

  ./macos-guest-virtualbox.sh documentation | less -R

Press enter to review the script configuration

vm_name="macOS"
macOS_release_name="Catalina"    # install "HighSierra" "Mojave" "Catalina"
storage_size=80000               # VM disk image size in MB. minimum 22000
storage_format="vdi"             # VM disk image file format, "vdi" or "vmdk"
cpu_count=2                      # VM CPU cores, minimum 2
memory_size=4096                 # VM RAM in MB, minimum 2048
gpu_vram=128                     # VM video RAM in MB, minimum 34, maximum 128
resolution="1280x800"            # VM display resolution

These values may be customized as described in the documentation.

Press enter to continue, CTRL-C to exit
Can't find VBoxManage in PATH variable,
checking C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
Found VBoxManage
Locally installing dmg2img
dmg2img-1.6.6-win32.zip                                             100%[==================================================================================================================================================================>]  62.87K   230KB/s    in 0.3s
Archive:  dmg2img-1.6.6-win32.zip
  inflating: dmg2img.exe

Catalina selected to be downloaded and installed

stage: prompt_delete_existing_vm

stage: create_vm

stage: check_default_virtual_machine

Checking that VirtualBox starts the virtual machine without errors.

Checking that VirtualBox uses hardware-supported virtualization.

stage: prepare_macos_installation_files

Downloading Apple macOS Catalina software update catalog
Catalina_sucatalog                                                  100%[==================================================================================================================================================================>]   6.30M  24.2MB/s    in 0.3s
Trying to find macOS Catalina InstallAssistant download URL
Catalina_sucatalog_00_InstallAssistantAuto.smd                      100%[==================================================================================================================================================================>]  39.99K  --.-KB/s    in 0.03s
Found download URL: http://swcdn.apple.com/content/downloads/41/56/061-94403-A_5AM6QB8TGK/ijdhki76d4rw19t579eblaqc26ngzlhu1p/

Downloading macOS installation files from swcdn.apple.com

Splitting the several-GB InstallESDDmg.pkg into 1GB parts because
VirtualBox hasn't implemented UDF/HFS VISO support yet and macOS
doesn't support ISO 9660 Level 3 with files larger than 2GB.
creating file 'Catalina_InstallESD.part00'
creating file 'Catalina_InstallESD.part01'
creating file 'Catalina_InstallESD.part02'
creating file 'Catalina_InstallESD.part03'
creating file 'Catalina_InstallESD.part04'
creating file 'Catalina_InstallESD.part05'
creating file 'Catalina_InstallESD.part06'
creating file 'Catalina_InstallESD.part07'

stage: create_nvram_files

stage: create_macos_installation_files_viso
Creating EFI startup script

Creating VirtualBox 6 virtual ISO containing the
installation files from swcdn.apple.com


stage: create_basesystem_virtual_disk
Converting BaseSystem.dmg to BaseSystem.img
./macos-guest-virtualbox.sh: line 649: dmg2img: command not found
Converting from raw image file="Catalina_BaseSystem.img" to file="Catalina_BaseSystem.vdi"...
VBoxManage.exe: error: Cannot open file "Catalina_BaseSystem.img": VERR_FILE_NOT_FOUND
Failed to create "Catalina_BaseSystem.vdi".
Exiting.
```

**Comment by myspaghetti** [comment]
Make sure the path to `dmg2img.exe` is in the `PATH` variable, or put it in the script's working directory.

**Comment by phobrla** [comment]
Are we talking about something on a certain line of the macos-guest-virtualbox.sh file, or the Windows environment variables?

**Comment by myspaghetti** [comment]
I believe Cygwin configures `bash` by default to inherit the Windows `PATH` variable but just put `dmg2img.exe` in `~/bin/` or `/usr/local/bin` if you're unsure.

By the way it appears the `PWD` variable is not set, are you running the script in an interactive bash shell?

**Comment by phobrla** [comment]
I think so, I'm running it in the Cygwin64 terminal; in what should I be running it?

**Comment by myspaghetti** [comment]
If `PWD` was set, the script would have used `dmg2img.exe` from the working directory; `bash` should be launched with the `-i` (interactive)  and `--login` options. When you enter `echo ${PWD}` it should output the current working directory.

At any rate if you put `dmg2img.exe` in `~/bin/` or `/usr/local/bin` the script should find `dmg2img`. Additionally I think I'll add a test that verifies `PWD` is set, since you're the second person to report this issue.

**Comment by phobrla** [comment]
OK, I'm now in the actual install environment. After I select the language, I'm selecting Disk Utility, but neither Terminal nor the Partition dialog are appearing.

**Comment by myspaghetti** [comment]
Did check if `PWD` was set? I'd like to know what `bash` sessions don't set `PWD` if you don't mind.

When the script is active you're not supposed to manually interact with the VM.

**Comment by myspaghetti** [comment]
Duplicate of  #283, fixed in version 0.96.6
